uint32 seq 
objectPos[] object_pos
---
uint32 seq
bool[] object_result # for each particle, there exists a corresponding result